tools/azsdk-cli/docs/specs/0-scenario-2.spec.md

```diff
@@ -0,0 +1,818 @@
+- Tear down temporary resources with `azsdk_package_test_resources_manage --action delete`
+- Switch to playback mode with `azsdk_package_test_mode_set --mode playback`
+- Run playback validation with `azsdk_package_run_tests` using refreshed [test recordings](#test-recording)
+
+**Success:**
+
+- Re-recordings update only impacted files
+- Playback runs confirm compatibility with refreshed [test recordings](#test-recording)
+
+### 7. Update Package/Docs/Metadata
+
+Unchanged from Scenario 1. Refer to [Scenario 1 – Update Package/Docs/Metadata](./0-scenario-1.spec.md#3-update-packagedocsmetadata) for complete details.
+
+**Success (additional for Scenario 2):**
+
+- Tooling correctly handles package metadata updates for both [Net-New SDKs](#net-new-sdk) and existing SDKs
+
+### 8. Validating
+
+Unchanged from Scenario 1. Refer to [Scenario 1 – Validating](./0-scenario-1.spec.md#4-validating) for validation steps and success criteria.
+
+**Success (additional for Scenario 2):**
+
+- Validation checks pass for both [Net-New SDKs](#net-new-sdk) and existing SDKs
+
+---
+
+## Success Criteria
+
+Scenario 2 is complete when:
+
+- The complete workflow executes successfully for all five languages
+- Both Agent Mode and CLI Mode scenarios run end to end
+- Documentation exists for both modes, reflecting new customization and testing stages
+- **Runs are repeatable**: CLI Mode produces deterministic results; Agent Mode may vary in execution path but MCP tools produce deterministic outputs given the same inputs
+- Agent prompts trigger expected tool sequences
+- CLI commands execute with expected outputs
+- Workflow runs entirely on the local machine
+- No changes are committed to repositories
+- Live-test recordings are generated or refreshed and playback validation succeeds
+- **All new tools function correctly** as defined in their respective stage details:
+  - `azsdk_package_setup_env` remediates environment issues
+  - `azsdk_package_customize_typespec` modifies and validates TypeSpec
+  - `azsdk_package_customized_code_update` applies code customizations
+  - `azsdk_package_customization_playbook` provides TypeSpec vs. code guidance
+  - `azsdk_package_run_tests` executes tests in current mode
+  - `azsdk_package_test_resources_manage` provisions/destroys Azure resources
+  - `azsdk_package_test_assets_create` generates test infrastructure files
+  - `azsdk_package_test_mode_set` switches testing modes
+- **Both [Net-New SDKs](#net-new-sdk) and existing SDKs** are fully supported throughout all stages
+
+---
+
+## Agent Prompts
+
+### Full Workflow (Net-New SDK)
+
+**Prompt:**
+
+```text
+Create a new Health Deidentification SDK across all languages: apply any needed customizations, generate code, add helper APIs, and run live tests with recordings then validate playback.
```

Comment:
what are helper APIs here? how do we know what customizations we want here?

Reply:
Good question. I now thinking due to the nature of both types of customizations, they would never really be a part of the "run the full process" workflow and will always be a specific thing that is worked on when specifically called out at that time in the process. 

I'll remove the customization parts from the full workflow prompt.